ddtrace/appsec/_iast/_taint_tracking/aspects.py

```diff
@@ -996,9 +995,9 @@ def re_finditer_aspect(
         string = args[0]
         if is_pyobject_tainted(string):
             ranges = get_ranges(string)
-            for elem in result:
+            result, result_backup = itertools.tee(result)
+            for elem in result_backup:
```

Comment:
Is this tainting `result` because the underlying pointers point to the same objects?

Reply:
Exactly